File: 1.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $nums1 = [];
        foreach ($nums as $k => $v) {
            $difference = $target - $v;
            if (isset($nums1[$difference])) {
                return [$k, $nums1[$difference]];
            } else {
                $nums1[$v] = $k;
            }
        }
    }
}

var_dump((new Solution)->twoSum([2, 7, 11, 15], 9));

File: 1/1.go
package main

import "fmt"

func main() {
	fmt.Print(twoSum([]int{1, 2, 3}, 3))
}
func twoSum(nums []int, target int) []int {
	intMap := make(map[int]int)
	for k, v := range nums {
		tmp := target - v
		if _, isset := intMap[tmp]; isset {
			return []int{k, intMap[tmp]}
		}
		intMap[v] = k
	}
	return nil
}
